onedal/decomposition/tests/test_incremental_pca.py

```diff
@@ -0,0 +1,187 @@
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_on_random_data(
+    queue, n_components, whiten, num_blocks, row_count, column_count, dtype
+):
+    seed = 78
+    gen = np.random.default_rng(seed)
+    X = gen.uniform(low=-0.3, high=+0.7, size=(row_count, column_count))
+    X = X.astype(dtype=dtype)
+    X_split = np.array_split(X, num_blocks)
+    incpca = IncrementalPCA(n_components=n_components, whiten=whiten)
+
+    for i in range(num_blocks):
+        incpca.partial_fit(X_split[i], queue=queue)
+
+    incpca.finalize_fit()
+
+    transformed_data = incpca.predict(X)
+    tol = 3e-3 if dtype == np.float32 else 2e-6
+
+    n_components = incpca.n_components_
+    expected_n_samples_seen = X.shape[0]
+    expected_n_features_in = X.shape[1]
+    n_samples_seen = incpca.n_samples_seen_
+    n_features_in = incpca.n_features_in_
+    assert n_samples_seen == expected_n_samples_seen
+    assert n_features_in == expected_n_features_in
+
+    components = incpca.components_
+    singular_values = incpca.singular_values_
+    centered_data = X - np.mean(X, axis=0)
+    cov_eigenvalues, cov_eigenvectors = np.linalg.eig(
+        centered_data.T @ centered_data / (n_samples_seen - 1)
+    )
+    cov_eigenvalues = np.nan_to_num(cov_eigenvalues)
+    cov_eigenvalues[cov_eigenvalues < 0] = 0
+    eigenvalues_order = np.argsort(cov_eigenvalues)[::-1]
+    sorted_eigenvalues = cov_eigenvalues[eigenvalues_order]
+    sorted_eigenvectors = cov_eigenvectors[:, eigenvalues_order]
+    expected_singular_values = np.sqrt(sorted_eigenvalues * (n_samples_seen - 1))[
+        :n_components
+    ]
+    expected_components = sorted_eigenvectors.T[:n_components]
+    assert_allclose(singular_values[:-1], expected_singular_values[:-1], atol=tol)
+
+    assert_allclose(singular_values, expected_singular_values, atol=tol)
+    for i in range(n_components):
+        abs_dot_product = np.abs(np.dot(components[i], expected_components[i]))
+        assert np.abs(abs_dot_product - 1.0) < tol
+
+    expected_explained_variance = sorted_eigenvalues[:n_components]
+    assert_allclose(incpca.explained_variance_, expected_explained_variance, atol=tol)
+
+    expected_explained_variance_ratio = expected_explained_variance / np.sum(
+        sorted_eigenvalues
+    )
+    assert_allclose(
+        incpca.explained_variance_ratio_, expected_explained_variance_ratio, atol=tol
+    )
+
+    expected_noise_variance = (
+        np.mean(sorted_eigenvalues[n_components:])
+        if len(sorted_eigenvalues) > n_components
+        else 0.0
+    )
+    # TODO Fix noise variance computation (It is necessary to update C++ side)
+    # assert np.abs(incpca.noise_variance_ - expected_noise_variance) < tol
+
+    expected_transformed_data = centered_data @ components.T
+    if whiten:
+        scale = np.sqrt(incpca.explained_variance_)
+        min_scale = np.finfo(scale.dtype).eps
+        scale[scale < min_scale] = np.infty
```

Comment:
```suggestion
        scale[scale < min_scale] = np.inf
```
if its equivalent - we use np.inf in other places in repo